packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -375,18 +378,19 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
 
   private _onRenderDescription = (props: IButtonProps) => {
     const {
-      description
+      description,
+      secondaryText
     } = props;
 
     // ms-Button-description is only shown when the button type is compound.
     // In other cases it will not be displayed.
-    return description ? (
+    return (secondaryText || description) ? (
```

Comment:
It seems like we should have a component props filter in createComponent... doing this should be much more clean.

Reply:
I thought so at first too, but the cases I've had to change have been relatively clean. As-is, removing the deprecated prop would just result in a few easy type errors to fix.